### PR TITLE
Rework and extends docs

### PR DESCRIPTION
**Pull request by boguslavsky**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,200 +1,513 @@
...
 ```html
-<script src="dist/exonum-client.min.js"></script>
-<script>
-    Exonum.hash([0, 255, 16, 8]);
-</script>
+<script src="node_modules/exonum-client/dist/exonum-client.min.js"></script>
 ```
 
-Usage in Node.js:
+Include in Node.js:
```
Comment (slowli) [inline README.md]: Nitpick: I like "Usage" more.

File: README.md [hunk 2]
```diff
@@ -1,200 +1,513 @@
...
-| fields | `Array` | yes | Array of fields |
+| Field | Description | Type |
+|---|---|---|
+| **type** | Definition of the field type. | [Built-in primitive](#built-in-primitives), [array](#arrays) or [custom data type](#nested-data-types) defined by the developer. | 
+| **size** | Total length of the field in bytes. | `Number` |
+| **from** | The beginning of the field segment in the byte array. | `Number` |
+| **to** | The end of the field segment in the byte array. | `Number` |
 
-#### Field structure:
+### Built-in primitives
```
Comment (slowli) [inline README.md]: Nitpick: "Built-in types" looks better.

File: README.md [hunk 3]
```diff
@@ -203,362 +516,795 @@ var SendFunds = Exonum.newMessage({
...
-| Uint16 | `number` | `2` | Unsigned integer in a range from `0` to `65535` |
-| Uint32 | `number` | `4` | Unsigned integer in a range from `0` to `4294967295` |
-| Uint64 | `number` or `string`* | `8` | Unsigned integer in a range from `0` to `18446744073709551615` |
+An example of proof of the existence of some data of `Wallet` type:
 
-*\*Note that JavaScript limits minimum and maximum integer number.
-Maximum safe integer in JavaScript is `2^53-1` which is equal to `9007199254740991`.
-For numbers greater than safe use `string` only.*
+```javascript
+// Proof from blockchaun
```
Comment (slowli) [inline README.md]: Typo: s/blockchaun/blockchain/

File: README.md [hunk 4]
```diff
@@ -203,362 +516,795 @@ var SendFunds = Exonum.newMessage({
...
+
+#### String to Uint8Array
+
+```javascript
+var str = 'Hello world';
+
+Exonum.stringToUint8Array(str); // [72, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100]
 ```
 
+## Community
```
Comment (slowli) [inline README.md]: IMO, this section does not belong to README; the links are present on the Exonum website.

File: README.md [hunk 5]
```diff
@@ -1,200 +1,513 @@
...
-    Exonum.hash([0, 255, 16, 8]);
-</script>
+<script src="node_modules/exonum-client/dist/exonum-client.min.js"></script>
 ```
 
-Usage in Node.js:
+Include in Node.js:
 
 ```javascript
 var Exonum = require('exonum-client');
```
Comment (slowli) [inline README.md]: Maybe, include an ES6 import statement as well? Will it work?

**Comment by boguslavsky** [comment]
> On the other hand, the code samples contribute to the length of the README, which IMO could make it inapproachable by external users. I wonder whether it could make sense to, say, reuse type and message definitions across examples - it could significantly reduce the size code samples.

I agree, the current length makes README a bit unusable. But I would rather propose to move all the large examples to internal README files. In my opinion, it will be easier to understand independent code blocks according some narrow topic, rather than search the answer among the entire README file.

**Comment by boguslavsky** [comment]
> I think this may be a good time to rewrite code samples in README using standard code style; right now, they are inconsistent (indentation, semicolons, use of var opposed to let / const, etc.).

@slowli Done.

**Comment by boguslavsky** [comment]
> - On the other hand, the code samples contribute to the length of the README, which IMO could make it inapproachable by external users. I wonder whether it could make sense to, say, reuse type and message definitions across examples - it could significantly reduce the size code samples.
> - Similarly, I think it could make sense to shorten long strings, Merkle proofs, etc. I don't see the role of the samples to be makeshift tests, so I think it would be acceptable if the example code doesn't compile (but it's understandable how to adapt it for real use cases).

@slowli can you please take a look on updated README. I made README ~ 2 times shorter by moving of long usage examples into separate markdown files.

**Pull request merged**